src/plumpy/process_comms.py

```diff
@@ -42,10 +44,45 @@ class Intent:
     STATUS: str = 'status'
 
 
-PAUSE_MSG = {INTENT_KEY: Intent.PAUSE}
-PLAY_MSG = {INTENT_KEY: Intent.PLAY}
-KILL_MSG = {INTENT_KEY: Intent.KILL}
-STATUS_MSG = {INTENT_KEY: Intent.STATUS}
+MessageType = Dict[str, Any]
+
+
+class PlayMessage:
+    @classmethod
+    def build(cls, message: str | None = None) -> MessageType:
+        return {
+            INTENT_KEY: Intent.PLAY,
+            MESSAGE_KEY: message,
+        }
+
+
+class PauseMessage:
+    @classmethod
+    def build(cls, message: str | None = None) -> MessageType:
+        return {
+            INTENT_KEY: Intent.PAUSE,
+            MESSAGE_KEY: message,
+        }
+
+
+class KillMessage:
+    @classmethod
+    def build(cls, message: str | None = None, force: bool = False) -> MessageType:
```

Comment:
note, aiida-core sends a kill message [here](https://github.com/aiidateam/aiida-core/blob/ec52f4ef321f9ef1fa24e5a3056153ea55bce7d4/src/aiida/cmdline/commands/cmd_process.py#L344C1) and then [here](https://github.com/aiidateam/aiida-core/blob/ec52f4ef321f9ef1fa24e5a3056153ea55bce7d4/src/aiida/engine/processes/control.py#L202), how the flag should be send over?

Reply:
For the one in `_perform_action`, it is the same, but I'd prefer to have the argument passing close to the function call of `kill_process`

```python
_perform_actions(processes, controller.kill_process, 'kill', 'killing', timeout, wait, msg=message)
```

change it to

```python
message = KillMessage.build(message=message, force=force) 
_perform_actions(processes, functool.partial(controller.kill_process, msg=message), 'kill', 'killing', timeout, wait)
```

Reply:
> but now aiida-core needs to import KillMessage from plumpy only for adding a flag

Why it is a problem for importing `KillMessage`? I think message is a good interface that can flexibly support backward compatibility. We make all four message type public APIs of plumpy and when using process communicator it requires sending over through this message. In the future even if we want to change the real message to other type we can do it without break users' code.

 >  wouldn't be easier that msg itself could be a dictionary?

It is a dictionary returned by `build` but construct a dictionary by hand is error-prone. Using the `build` function of the message, the function signature will tell developers which options are allowed to be passed. We can make this works also by dataclass, but then that is overkill for a message in my opinion.

Reply:
well this the twisted things that we were always complaining about, import from plumpy to only to make a dictionary only for an input of a plumpy function itself... 
I mean you see my point :smile: :pinched_fingers:

Reply:
Sorry @sphuber, I just saw your comment, I had my browser open for during the weekend and I directly reply to Ali's comment without refresh.

Yes, thought about dataclass, and I think it can work. I use `build` into a dict mostly for the reason to keep the message the same as what was passing as before, since this is the message that should also go through de/serialize by the en/decoder and I am not 100% confident with that.

I think msgpack will take care of the dataclass for sure, but since the en/decoder used was the yaml one, I am not so sure.